### ISSUE TITLE
CORS error for tracmip csv

### ISSUE THREAD

**Issue by rabernat** [pangeo-data/pangeo-datastore-flask#22]
On http://catalog.pangeo.io/climate/tracmip, I saw this in my javascript console
```
Access to XMLHttpRequest at 'https://storage.googleapis.com/cmip6/tracmip.csv' from origin 'http://catalog.pangeo.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

We fixed this once before for CMIP6. Do we have any record of the commands used to fix it?

**Comment by charlesbluca** [comment]
The way we fixed it before was by adding `Access-Control-Allow-Origin : '*'`, but it may have been easier since the CSV file was located in a Pangeo bucket. This time around we might be able to set the CORS header using `gsutil cors set`, but I'm not sure if that can be done on a file level as documentation seems to set CORS headers at the bucket level.

**Comment by charlesbluca** [comment]
Like I thought, there doesn't seem to be a reliable way to set CORS on the CSV file without changing it for the entire bucket. I think a safer option for now would be to go the route of the CMIP6 data - host the CSV in the pangeo-cmip6 bucket which has the appropriate CORS settings.

**Comment by charlesbluca** [comment]
Got a temporary solution running by moving the CSV to pangeo-cmip6; if or when we want to move these catalogs to Google's CMIP6 bucket, the CORS option we have set on the pangeo-cmip6 bucket is:

`[{"maxAgeSeconds": 3600, "method": ["GET", "HEAD"], "origin": ["*"], "responseHeader": ["Content-Type"]}]`

**Comment by rabernat** [comment]
Did you try just setting it on the one file? I don’t think you have to apply it to the whole bucket.

**Comment by charlesbluca** [comment]
The documentation specifies that CORS commands (at least from `gsutil`) can only be used on buckets, not objects. When inspecting why the CMIP6 CSV file has no problems loading, it appears that the CORS options we set are on the bucket itself, and there doesn't seem to be anything changed about the metadata of the individual file.

**Comment by rabernat** [comment]
> When inspecting why the CMIP6 CSV file has no problems loading, it appears that the CORS options we set are on the bucket itself,

So if the TRACMIP CSV in the same bucket, shouldn't it work as well?

**Comment by rabernat** [comment]
```
$ gsutil cors get gs://cmip6
gs://cmip6/ has no CORS configuration.
```

So why does the CMIP6 csv work?

**Comment by charlesbluca** [comment]
The CMIP6 ESMCol JSON file actually links to a CSV file located in pangeo-cmip6 - this is where the proper CORS configuration is set, allowing the website to access the remote file.

**Comment by rabernat** [comment]
Ok I finally understand!

https://storage.googleapis.com/pangeo-cmip6/pangeo-cmip6-zarr-consolidated-stores.csv
https://storage.googleapis.com/cmip6/cmip6-zarr-consolidated-stores.csv

We are storing the catalog twice. This seems like a big opportunity for problems. Does @naomi-henderson know about this?

I think we should probably apply the CORS policy to the CMIP6 bucket and close down the pangeo-cmip6 bucket. Can anyone see a problem with this idea? Is there any downside to setting the CORS policy on the CMIP6 bucket?

**Comment by charlesbluca** [comment]
None that I know of, as long as we limit access to GET and HEAD as we've been doing for gs://pangeo-cmip6. If @naomi-henderson is sure it won't cause any problems with any of the other data there, I think it would be a great idea to consolidate it all and close the pangeo-cmip6 bucket.

**Comment by charlesbluca** [comment]
Closing this now that we've resolved the CORS issues on `gs://cmip6`; we can open up a new issue if we need to further refine CORS on this bucket.
